File: olympics/open/1.py
x = 1e8 - 400

c = 0
for p in range(2, 1000, 3):
    if (c + (1 << p)) >= x:
        for i in range(1 << p, 1 << (p + 1)):
            c += 1
            if c == x:
                print(i)
                break
    else:
        c += 1 << p

File: cntrea/ai/networks/1.go
// package main

// type NodeIndex int

// type Node struct {
// 	op       func([]float64) float64
// 	children []NodeIndex
// }

// type Network struct {
// 	nodes *[]Node
// }

func (net Network) NewNode(op func([]float64) float64, children []NodeIndex) NodeIndex{
	var cldn []NodeIndex
	n := NodeIndex(len(*net.nodes))
	
	for _, u := range children{
		if (u >= n){
			panic(1)
		}else{
			cldn = append(cldn, u)
		}
	}
	
	*net.nodes = append(*net.nodes, Node{op, cldn})

	return n
}

func NewNetwork() Network{
	var nodes []Node
	return Network{&nodes}
}
